File: Java/competitions/cd.java
import java.io.*;
import java.util.*;

public class cd {
	public static int indexOf(int[] a, int key) {
        int lo = 0;
        int hi = a.length - 1;
        while (lo <= hi) {
            // Key is in a[lo..hi] or not present.
            int mid = lo + (hi - lo) / 2;
            if      (key < a[mid]) hi = mid - 1;
            else if (key > a[mid]) lo = mid + 1;
            else return mid;
        }
        return -1;
    }
	public static void main(String[] args) throws IOException{
		BufferedReader kb = new BufferedReader(new InputStreamReader(System.in));
		while(true){
			String[] numcds = (kb.readLine()).split(" ");
			int[] numCds = {Integer.parseInt(numcds[0]), Integer.parseInt(numcds[1])};
			if(numCds[0]==0 && numCds[1]==0){
				break;
			}else{
				int[] jack = new int[numCds[0]];
				int total = 0;
				for(int x=0;x<jack.length;x++){
					jack[x]=Integer.parseInt(kb.readLine());
				}
				Arrays.sort(jack);
				for(int x=0;x<numCds[1];x++){
					int num = Integer.parseInt(kb.readLine());
					if(indexOf(jack, num)>-1){
						total++;
					}
				}

				System.out.println(total);
			}
			
		}
	}
}

File: Python/cd.py
while True:
    numcds = input().split(" ")
    numcds[0] = int(numcds[0]); numcds[1] = int(numcds[1])
    if numcds[0] == 0 and numcds[1] == 0:
        break
    else:
        jack =[]
        jill = 0
        for x in range(0,numcds[0]):
            jack.append(int(input()))
        for y in range(0,numcds[1]):
            num = int(input())
            if num in jack:
                jill += 1
        print(jill)